feat: add security and reliability improvements

alfonsodg opened the pull request:
## Summary
This PR adds several security and reliability improvements to the mem0-mcp-server.

## Changes

### Security Improvements
- ✅ **API Key Validation** - Validates Mem0 API key format (m0- prefix and minimum length)
- ✅ **DNS Rebinding Protection** - Enabled by default, configurable via `MEM0_DISABLE_DNS_REBINDING_PROTECTION`

### Reliability Improvements
- ✅ **LRU Cache** - Implements LRU cache with 100 client limit to prevent memory leaks
- ✅ **Retry Logic** - Exponential backoff for 5xx errors (3 retries: 1s, 2s, 4s delays)

### Configuration Enhancements
- ✅ **CLI Arguments** - Support for `--api-key` and `--user-id` arguments for clients that don't support env vars
- ✅ **Priority Order** - CLI args > env vars for flexible configuration

### Code Quality
- ✅ **Refactoring** - Removed unnecessary import fallback
- ✅ **Tests** - Added 13 unit tests covering core logic
- ✅ **.gitignore** - Added for Python artifacts

## Testing
```bash
pytest tests/ -v
# 13 passed in 0.43s ✓
```

## Configuration Examples

**With environment variables:**
```json
{
  "mem0": {
    "command": "uvx",
    "args": ["mem0-mcp-server"],
    "env": {
      "MEM0_API_KEY": "m0-...",
      "MEM0_DEFAULT_USER_ID": "user-id"
    }
  }
}
```

**With CLI arguments:**
```json
{
  "mem0": {
    "command": "uvx",
    "args": [
      "mem0-mcp-server",
      "--api-key=m0-...",
      "--user-id=user-id"
    ]
  }
}
```

## Issues Closed
- Closes #1 - API key validation
- Closes #2 - LRU cache implementation
- Closes #3 - DNS rebinding protection
- Closes #6 - Retry logic
- Closes #7 - Import fallback cleanup

## Backward Compatibility
All changes are backward compatible. Existing configurations will continue to work without modifications.

alfonsodg committed:
refactor: remove unnecessary import fallback

Closes #7

alfonsodg committed:
feat: add security and reliability improvements

- Validate API key format (m0-prefix)
- Implement LRU cache with 100 client limit
- Enable DNS rebinding protection by default
- Add retry logic with exponential backoff for 5xx errors
- Document new MEM0_DISABLE_DNS_REBINDING_PROTECTION env var

Closes #1
Closes #2
Closes #3
Closes #6

alfonsodg committed:
chore: add .gitignore for Python artifacts

alfonsodg committed:
feat: add CLI argument support for API key and user ID

Allows configuration via command-line arguments for MCP clients
that don't support environment variables.

Usage:
  mem0-mcp-server --api-key=m0-... --user-id=your-handle

CLI arguments take precedence over environment variables.

alfonsodg committed:
test: add comprehensive test suite

- API key validation tests
- Default filter injection tests
- LRU cache behavior tests
- Retry logic with exponential backoff tests
- CLI argument parsing tests

Coverage:
- Core server functionality
- Error handling and retries
- Configuration options

alfonsodg committed:
test: simplify tests to avoid external dependencies

- 13 tests covering core logic validation
- API key format validation
- Filter injection logic
- CLI argument parsing
- Retry logic calculations
- Error code detection

All tests passing: 13/13 ✓

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mem0ai/mem0-mcp?pullRequest=26) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mem0ai/mem0-mcp?pullRequest=26) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mem0ai/mem0-mcp?pullRequest=26) it.</sub>

alfonsodg committed:
chore: rename package to mem0-mcp-server-alfonsodg v0.2.2

Allows installation alongside official package until PR is merged.

alfonsodg committed:
docs: add provisional deployment section with GitHub URL

Shows how to deploy directly from GitHub without installation
using uvx with placeholder for API key.

alfonsodg committed:
docs: explain why CLI args are needed instead of env vars

GitHub Copilot CLI has issues with environment variables in MCP
server configurations. CLI arguments ensure compatibility.

alfonsodg committed:
fix: suppress warning when API key is provided via CLI args

The warning about missing MEM0_API_KEY now checks both environment
variable and CLI argument before displaying.

alfonsodg committed:
chore: bump version to 0.2.3

whysosaket commented:
This repository has been archived. `mem0-mcp-server` is no longer actively maintained — Mem0 now offers an official cloud-hosted MCP server that replaces this package.

**Get started with the new Mem0 MCP:** https://docs.mem0.ai/platform/mem0-mcp

Closing this as part of the archive. Thank you for your contribution!
